scaletest/templates/scaletest-runner/startup.sh

```diff
@@ -23,22 +23,56 @@ fi
 
 annotate_grafana "workspace" "Agent running" # Ended in shutdown.sh.
 
+{
+	pids=()
+	ports=()
+	declare -A pods=()
+	next_port=6061
+	for pod in $(kubectl get pods -l app.kubernetes.io/name=coder -o jsonpath='{.items[*].metadata.name}'); do
+		maybedryrun "${DRY_RUN}" kubectl -n coder-big port-forward "${pod}" "${next_port}:6060" &
+		pids+=($!)
+		ports+=("${next_port}")
+		pods[${next_port}]="${pod}"
+		next_port=$((next_port + 1))
+	done
+
+	trap 'trap - EXIT; kill -INT "${pids[@]}"; exit 1' INT EXIT
+
+	while :; do
+		sleep 285 # ~300 when accounting for profile and trace.
+		log "Grabbing pprof dumps"
+		start="$(date +%s)"
+		annotate_grafana "pprof" "Grab pprof dumps (start=${start})"
+		for type in allocs block heap goroutine mutex profile?seconds=10 trace?seconds=5; do
+			for port in "${ports[@]}"; do
+				maybedryrun "${DRY_RUN}" curl -sSL --output "${SCALETEST_PPROF_DIR}/pprof-${type}-${pods[${port}]}-${start}.gz" "http://localhost:${port}/debug/pprof/${type}"
```

Comment:
`?=` is annoying to deal with in a filename, might be good to strip that when creating.

Reply:
Good idea, I never have trouble with these due to my Zsh setup.